core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/ConsulMetadataSetup.java

```diff
@@ -58,6 +62,7 @@ public void setupConsulMetadata(Long stackId) {
         }
         updateWithConsulData(allInstanceMetaData, Collections.<InstanceMetaData>emptySet());
         instanceMetaDataRepository.save(allInstanceMetaData);
+        stackUpdater.updateStackStatus(stackId, Status.AVAILABLE, "Stack is ready");
```

Comment:
I think it would be better to update the Stack status in an upper layer (like in the facades) and do not leave it to the implementations.